core/src/main/java/org/jboss/pnc/build/finder/core/BuildFinder.java

```diff
@@ -595,11 +596,15 @@ private KojiBuild findBestBuildFromCandidates(List<KojiBuild> candidates, List<K
                     buildCache.put(cacheRpmBuildInfo.getBuildInfo().getId(), cacheRpmBuildInfo);
                 }
             } else {
-                if (cacheManager == null || (cacheArchiveInfos = checksumCaches.get(ChecksumType.md5)
-                        .get(checksum.getValue())) == null) {
+                ListKojiArchiveInfoProtobufWrapper wrapper = null;
+                if (checksumCaches != null && checksumCaches.containsKey(ChecksumType.md5)) {
```

Comment:
I am trying to remember why I  previously could do a `get` here without a `containsKey`. I think it creates an empty map for all possible checksum types earlier, so you may not need it. Plus, I always found `containsKey` to be redundant if you're doing a `get` that returns `null`.

Reply:
sure